Dockerfile.builder

```diff
@@ -0,0 +1,34 @@
+# Docker environment to build matching go-ipfs and plugin binaries.
+# This will compile go-ipfs v0.12.0 and the current sources of our plugin to match that.
+# Sources are placed in /usr/src/ipfs/.
+# Artifacts are put in /usr/local/bin/ipfs/.
+
+# We need go v1.16 because of a weird compiler bug in 1.17.
+# Also, we want bullseye as the build environment in order to use a somewhat old libc.
+# This improves compatibility with older host systems at no loss of functionality.
+FROM golang:1.16-bullseye AS builder
+
+# First, get and compile go-ipfs v0.12.0.
+# We need matching go-ipfs and plugin executables, so it makes sense to build them together.
+# We build go-ipfs first, because, since we're building off a defined tag, the sources don't change and this can be
+# cached.
+WORKDIR /usr/src/ipfs/go-ipfs
+RUN git clone https://github.com/ipfs/go-ipfs.git . && git checkout v0.12.0
+RUN go build -v -o /usr/local/bin/ipfs/ipfs-v0.12.0-docker ./cmd/ipfs
+
+# Compile metric export plugin
+WORKDIR /usr/src/ipfs/metric-export-plugin
+# Get and compile dependencies first, since we can cache those.
+COPY go.mod go.sum ./
+RUN go mod download && go mod verify
+
+# Then copy sources and compile project
+COPY . .
```

Comment:
This feels a bit wonky but I guess just copying all the things does no real harm either